### PR TITLE
fill in readme file

### PR DESCRIPTION
**Pull request by samccann**
##### SUMMARY

Add more details to the README.md file

##### ISSUE TYPE

- Docs Pull Request

##### COMPONENT NAME

documentation
##### ADDITIONAL INFORMATION

```paste below

```

### CODE REVIEW

**Comment by trishnaguha** [comment]
@cidrblock @ganeshrn @Qalthos Will you please review the PR?

File: README.md [hunk 1]
```diff
@@ -0,0 +1,73 @@
+
+
+# Ansible Network Collection for Common Code (netcommon)
+[![CI](https://zuul-ci.org/gated.svg)](https://dashboard.zuul.ansible.com/t/ansible/builds?project=ansible-collections%2Fansible.netcommon) [![Codecov](https://img.shields.io/codecov/c/github/ansible-collections/ansible.netcommon)](https://codecov.io/gh/ansible-collections/ansible.netcommon)
+
+The Ansible ``ansible.netcommon`` collection includes common network content to help automate the management of network devices.
```
Comment (ganeshrn) [inline README.md:6]: The httpapi connection plugin is used by security and VMware modules as well so many this line can be more generic

File: README.md [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+This includes  connection plugins, such as ``httpapi`` and ``netconf``.
```
Comment (ganeshrn) [inline README.md:7]: Also mention `network_cli`

File: README.md [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+
+## Included content
+
+This collection includes:
+- Common network modules, such as ``net_*``, ``cli_*``, ``netconf_*``, and ``restconf_*`` modules.
+- Common connection plugins, such ``httpapi`` and ``network_cli``.
```
Comment (ganeshrn) [inline README.md:13]: Also netconf :)

File: README.md [hunk 4]
```diff
@@ -0,0 +1,73 @@
...
+- ``ipadrr`` and ``network`` filters.
```
Comment (ganeshrn) [inline README.md:14]: There are more filters. Should all of them be lister here https://github.com/ansible-collections/ansible.netcommon/blob/master/plugins/filter/network.py#L520-L528

https://github.com/ansible-collections/ansible.netcommon/blob/master/plugins/filter/ipaddr.py#L1157-L1177

File: README.md [hunk 5]
```diff
@@ -0,0 +1,73 @@
...
+
+```yaml
+---
+collections:
+  - name: ansible.netcommon
+    version: 0.0.3
+```
+## Using this collection
+
+The most common use case for this collection is to include it as a dependency in a network device-specific collection. See the [Vyos collection](https://github.com/ansible-collections/vyos) for an example of this.
```
Comment (ganeshrn) [inline README.md:34]: Maybe more information is required on how to use the plugins in these collections. Like using the fully qualified collection name for the connections for eg. `ansible.netcommon.network_cli`

File: README.md (hunk 4 above)
Comment (samccann) [reply README.md:14]: We have some documentation on these on docs.ansible.com An example is:
{{ output | parse_cli('path/to/spec') }}

Would that need to change to:
{{ output | ansible.necommon.parse_cli('path/to/spec') }} ?

**Comment by samccann** [comment]
I'll need to add a link to the 'how to contribute to Ansible maintained collections' page once that's available as well.

**Comment by trishnaguha** [comment]
@ganeshrn @samccann Is there any other blocker except for linking to the doc 'how to contribute to Ansible maintained collections'?

**Comment by samccann** [comment]
@trishnaguha no blocker but the code coverage badge will be broken because we don't have this collection up at:
https://codecov.io/gh/ansible-collections.  

So I'd like to have a decision on that one - will we put our collections up there or not? If not, I can remove that link.

File: README.md [hunk 6]
```diff
@@ -0,0 +1,114 @@
...
+- [Ansible Collection overview](https://github.com/ansible-collections/overview)
+- [Ansible User guide](https://docs.ansible.com/ansible/latest/user_guide/index.html)
+- [Ansible Developer guide](https://docs.ansible.com/ansible/latest/dev_guide/index.html)
+- [Ansible Community code of conduct](https://docs.ansible.com/ansible/latest/community/code_of_conduct.html)
+
+## Licensing
+
+GNU General Public License v3.0 or later.
+
+See [LICENCE](https://www.gnu.org/licenses/gpl-3.0.txt) to see the full text.
```
Comment (samccann) [inline README.md:114]:
```suggestion
See [LICENSE](https://www.gnu.org/licenses/gpl-3.0.txt) to see the full text.
```

File: README.md [hunk 7]
```diff
@@ -0,0 +1,114 @@
...
+
+### See Also:
+
+* [Ansible Using collections](https://docs.ansible.com/ansible/latest/user_guide/collections_using.html) for more details.
+
+## Contributing to this collection
+
+We welcome community contributions to this collection. If you find problems, please open an issue or create a PR against the [ansible.netcommon collection repository](https://github.com/ansible-collections/ansible.netcommon).
+
+You cal also join us on:
```
Comment (samccann) [inline README.md:86]:
```suggestion
You can also join us on:
```

**Comment by samccann** [comment]
removed codecov for now. Assuming it passes tests, this should be ready to go.

**Comment by JPrause** [comment]
> removed codecov for now. Assuming it passes tests, this should be ready to go.

@samccann I think this was the way to go for now.
We can always add this later if/when that becomes available.

**Pull request merged**